build_modules/lib/src/meta_module_clean_builder.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:collection';
+import 'dart:convert';
+
+import 'package:graphs/graphs.dart';
+
+import 'package:build/build.dart';
+import 'meta_module.dart';
+import 'meta_module_builder.dart';
+import 'modules.dart';
+
+/// The extension for serialized clean meta module assets.
+const metaModuleCleanExtension = '.meta_module_clean';
+
+/// Map of [AssetId] to corresponding primary [AssetId] within the same
+/// [Module].
+final _assetToModule =
+    new Resource<Map<AssetId, Module>>(() => {}, dispose: (map) => map.clear());
+
+/// Returns a set of all modules transitvely reachable from the provided meta
+/// module asset.
+Future<Set<Module>> _transitiveModules(BuildStep buildStep, AssetId metaAsset,
```

Comment:
nit: I prefer to have private implementations below the public apis in the file. Also I think its a bit more natural to put private methods implementations below their actual usage in the file, roughly in the order in which they are invoked by the public methods.